Assets/MixedRealityToolkit.Extensions/Sharing/SpatialAlignment/Common/ISpatialCoordinate.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Numerics;
+
+namespace Microsoft.MixedReality.Experimental.SpatialAlignment.Common
+{
+    /// <summary>
+    /// Defines the potential states of alignment for a strategy.
+    /// </summary>
+    public enum LocatedState
+    {
+        /// <summary>
+        /// An error has occurred with the alignment strategy.
+        /// </summary>
+        Error = -1,
+
+        /// <summary>
+        /// Alignment has never been resolved in this session.
+        /// </summary>
+        Unresolved = 0,
+
+        //TODO this vs Resolved, what are the use cases.
+        /// <summary>
+        /// Alignment is temporarily inhibited (usually due to some form of tracking loss).
+        /// </summary>
+        Inhibited,
+
+        /// <summary>
+        /// Alignment has been resolved at least once but there is no active tracking.
+        /// </summary>
+        Resolved,
+
+        /// <summary>
+        /// Alignment has been resolved at least once and there is active tracking.
+        /// </summary>
+        Tracking,
+    }
+
+    /// <summary>
+    /// This represents a spatial coordinate that can then be used to convert position and rotation to and from this coordinate space.
+    /// </summary>
+    public interface ISpatialCoordinate
+    {
+        /// <summary>
+        /// Occurs when the value of the <see cref="State"/> property has changed.
+        /// </summary>
+        event Action StateChanged;
+
+        /// <summary>
+        /// Gets the Id representing this coordinate.
+        /// </summary>
+        string Id { get; }
```

Comment:
Does the Id need to be a string? Why not a unique int / uid?
It could have a name as well, but I think the core unique id should be an int for fast comparisons.

Reply:
There's a lot of precident for a string ID. Azure Mobile Services ended up going with string and the new Cloud Spatial Anchors [uses](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.spatialanchors.cloudspatialanchor.identifier#Microsoft_Azure_SpatialAnchors_CloudSpatialAnchor_Identifier) string.

I know strings aren't fast, but they're flexible. A string can represent an int, a GUID, or really any type of key. This provides some flexibility in how the coordinate abstraction system maps into various underlying stores.

Reply:
They're flexible but they spit out garbage every time you use them. I understand the precedent, but I still think an int is a better choice.

You can hash other keys and store them as an int.

Regardless, I understand this isn't changing.